src/ui/public/agg_types/param_types/field.js

```diff
@@ -79,7 +79,12 @@ export function FieldParamTypeProvider(Private, $filter) {
 
     const validField = this.getFieldOptions(aggConfig).byName[fieldName];
     if (!validField) {
-      notifier.error(`Saved "field" parameter is now invalid. Please select a new field.`);
+      // check to see if a .raw field exists and use it if it does
+      const validRawField = this.getFieldOptions(aggConfig).byName[`${fieldName}.raw`];
+      if (validRawField) {
+        return validRawField;
```

Comment:
I think we should probably display an "info" notification when we change the field, so the user is aware of what's going on. It's possible they're intentionally trying to aggregate on a `text` field and just forgot to turn field data on, so without a notification that the field changed they may never realize they're not getting what they want.